testsuite/model/src/test/java/org/keycloak/testsuite/model/infinispan/SplitBrainTest.java

```diff
@@ -0,0 +1,100 @@
+package org.keycloak.testsuite.model.infinispan;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.keycloak.connections.infinispan.InfinispanConnectionProvider.WORK_CACHE_NAME;
+
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+import org.keycloak.connections.infinispan.InfinispanConnectionProvider;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RealmProvider;
+import org.keycloak.models.UserProvider;
+import org.keycloak.models.UserSessionProvider;
+import org.keycloak.models.cache.CacheRealmProvider;
+import org.keycloak.models.session.UserSessionPersisterProvider;
+import org.keycloak.testsuite.model.KeycloakModelTest;
+import org.keycloak.testsuite.model.RequireProvider;
+
+@RequireProvider(CacheRealmProvider.class)
+@RequireProvider(InfinispanConnectionProvider.class)
+@RequireProvider(RealmProvider.class)
+@RequireProvider(UserSessionPersisterProvider.class)
+@RequireProvider(UserSessionProvider.class)
+@RequireProvider(UserProvider.class)
+public class SplitBrainTest extends KeycloakModelTest {
```

Comment:
Please provide a bit more context what kind of split brains we consider here, and what the expected behavior of Keycloak should be. Put this as JavaDoc either on the class or the method, or split it between the two.

Reply:
I've add some Javadocs and renamed the test to `InfinispanEmbeddedSplitBrainTest` to make the purpose more explicit.